src/features/common/Accordion.tsx

```diff
@@ -0,0 +1,53 @@
+import * as Accordion from "@radix-ui/react-accordion";
+import { MdChevronRight } from "react-icons/md";
+import { twJoin } from "tailwind-merge";
+import { ReactNode, useState } from "react";
+
+interface AccordionComponentProps {
+  rootStyles: string;
+  type?: "single" | "multiple";
+  itemId: string;
+  trigger: ReactNode;
+  content: ReactNode;
+}
+
+const AccordionComponent = ({
+  rootStyles,
+  type = "multiple",
+  itemId,
+  trigger,
+  content,
+}: AccordionComponentProps) => {
+  const [isAccordionOpen, setIsAccordionOpen] = useState<boolean>(false);
+  const toggleChevronIconUp = () =>
+    setIsAccordionOpen((isAccordionOpen) => !isAccordionOpen);
+
+  return (
+    <Accordion.Root className={rootStyles} type={type}>
+      <Accordion.Item className="flex flex-col overflow-hidden" value={itemId}>
+        <Accordion.Header>
+          <div
+            className={twJoin([
+              "flex border border-border-dark p-3",
+              "hover:bg-border-darkGray",
+            ])}
+          >
+            {trigger}
+            <Accordion.Trigger className={"align-end align-end w-fit"}>
+              <div onClick={toggleChevronIconUp}>
+                {!isAccordionOpen ? (
+                  <MdChevronRight size={32} className={"rotate-90"} />
```

Comment:
Would probably have set the class conditionally but this achieves the same.

Interested to know it `transition-transform` (perhaps with `duration-150` or something short-ish?) looks decent on this or if it would need a bit of work.

Reply:
I added the transition you mentioned. It looks a bit fidgety to me, but we should probably discuss it with designers. Can fix it in a future PR